### ISSUE TITLE
When freezing the model parameters and training only the newly added nn.Module, the loss becomes NaN

### ISSUE THREAD

**Issue by Travis-Tang-chen** [wutaiqiang/MoSLoRA#5]
I first freeze the model parameters:

`for param in self.parameters():
    param.requires_grad = False
`
Then, I unfreeze the parameters of the new layer that I want to train:

`for param in new_layer.parameters():
    param.requires_grad = True`

Whenever the model performs gradient updates, the loss and related gradients become NaN. Have you encountered this situation before?

Thanks a lot.

**Comment by wutaiqiang** [comment]
I did not encounter this situation before. You want to update the original linear layer at the same time?

**Comment by tangg555** [comment]
Thank you for your response! :) Actually, I want to insert a new linear layer between the frozen LLM layers. I'm wondering if the NaN values occur because the frozen layers are hindering the backpropagation of the loss. :/

**Comment by wutaiqiang** [comment]
Maybe. I suggest you debug via tracking the grads for each parameter.
